subprojects/docs/src/docs/userguide/dependencyLocking.adoc

```diff
@@ -0,0 +1,129 @@
+
+++++
+<sample id="locking-all" dir="userguide/dependencyManagement/dependencyLocking/lockingAllConfigurations" title="Locking all configurations">
+    <sourcefile file="build.gradle" snippet="locking-all"/>
+</sample>
+++++
+
+
+[NOTE]
+====
+Only configurations that can be resolved will have a lock file attached to them. Applying locking on non resolvable-configurations is simply a no-op.
+====
+
+=== Lock file location and format
+
+Lockfiles will be located in the folder `gradle/dependency-locks` inside the project or subproject directory.
+Each file is named by the configuration it locks and has the `lockfile` extension.
+
+The content of the file is a module notation per line, with a header giving some context.
+Module notations are ordered alphabetically, to ease diffs.
+
+++++
+<sample id="locking-all" dir="userguide/dependencyManagement/dependencyLocking/lockingSingleConfiguration" title="Lockfile content">
+    <sourcefile file="gradle/compileClasspath.lockfile"/>
+</sample>
+++++
+
+=== Running a build with a lockfile present
+
+The moment a build needs to resolve a configuration that has locking enabled and it finds a matching lockfile,
+it will use it to verify that the given configuration still resolves the same versions.
+The validation is as follows:
+
+* Existing entries in the lockfile must be matched in the build
+** A version mismatch or missing resolved module causes a build failure
+* Resolution result must not contain extra dependencies compared to the lock file content
+
+=== Generating and updating lockfiles
+
+In order to generate or update a lockfile, you specify the `--write-locks` command line argument in addition to the normal tasks that would trigger configurations to be resolved.
+This will cause the creation of a lockfile for each resolved configuration in that build execution.
+Note that if a lockfile existed previously, it is overwritten.
+
+=== Controlled lock file update
+
+In order to update only specific modules of a configuration, you can use the `--update-locks` command line flag.
+It takes a comma (`,`) separated list of module notations.
+In this mode, the existing lock file is still used as input to resolution, filtering out the modules targeted by the update.
+Note however that the resolution may cause other module versions to update, as dictated by the Gradle resolution rules.
+
+----
+❯ ./gradlew classes --update-locks org:foo,org:bar
+----
+
+Wildcards, indicated with `*`, can be used in the group or module name. They can be the only character or appear at the end of the group or module respectively.
+The following wildcard notations are valid:
+
+* `org*:foo`
+* `org:*`
+* `*:bar`
+* `org*:bar*`
+
+=== Deleting a lockfile
+
+Simply remove the file matching the configurations where you no longer want locking.
+
+=== Locking limitations
+
+It is currently not possible to lock the `classpath` configuration used for build plugins.
+
+=== Nebula locking plugin
```

Comment:
:x: Let's remove this section. We don't mention external plugins in it, as it may vary over time.

Reply:
It was referenced from another section, so I though it was a good idea to move it here. And also a way of indicating that there is not a 100% match between the two different feature sets.

Reply:
Right, I wasn't aware we mentioned it. I'd still be in favor of removing.